### ISSUE TITLE
Support GT, GTE, LT, LTE to String fields

### ISSUE THREAD

**Issue by noureldin-khaled** [neo4j/graphql#1462]
I currently have a use case where I want to compare a String field with GT, GTE, LT, LTE. Here is my Schema
```graphql
type Address{
    uuid: String
    full_address: String
    city: String
    state: String
    zip: String
    creation_date: String
}
```
As you can see I have a `creation_date` which I do store as a string of the format `yyyy-mm-dd H:m:s` (ex: `2022-05-23 16:23:40`). This allows me to compare using basic `>`, `>=`, `<`, `<=` operators to get nodes that are created after/before a certain date.
However, I can't do that as the library doesn't support GT, GTE, LT, LTE on the `creation_date` field, it gives an error when trying to use them
```graphql
query Addresses {
  addresses(where: { creation_date_GTE: "2020-01-01" }) {
    uuid
    full_address
    creation_date
  }
}
```
gives
```
{
  "errors": [
    {
      "message": "Field \"creation_date_GTE\" is not defined by type \"AddressWhere\". Did you mean \"creation_date_IN\", \"creation_date_NOT\", \"creation_date\", \"creation_date_NOT_IN\", or \"creation_date_CONTAINS\"?",
      "extensions": {
        "code": "GRAPHQL_VALIDATION_FAILED",
        "exception": {
          "stacktrace": [
            "GraphQLError: Field \"creation_date_GTE\" is not defined by type \"AddressWhere\". Did you mean \"creation_date_IN\", \"creation_date_NOT\", \"creation_date\", \"creation_date_NOT_IN\", or \"creation_date_CONTAINS\"?",
            "    at ObjectField (/var/www/graphql.zlien.com/node_modules/graphql/validation/rules/ValuesOfCorrectTypeRule.js:87:11)",
            "    at Object.enter (/var/www/graphql.zlien.com/node_modules/graphql/language/visitor.js:301:32)",
            "    at Object.enter (/var/www/graphql.zlien.com/node_modules/graphql/utilities/TypeInfo.js:391:27)",
            "    at visit (/var/www/graphql.zlien.com/node_modules/graphql/language/visitor.js:197:21)",
            "    at validate (/var/www/graphql.zlien.com/node_modules/graphql/validation/validate.js:91:24)",
            "    at validate (/var/www/graphql.zlien.com/node_modules/apollo-server-core/dist/requestPipeline.js:187:39)",
            "    at processGraphQLRequest (/var/www/graphql.zlien.com/node_modules/apollo-server-core/dist/requestPipeline.js:90:34)",
            "    at processTicksAndRejections (node:internal/process/task_queues:96:5)",
            "    at async processHTTPRequest (/var/www/graphql.zlien.com/node_modules/apollo-server-core/dist/runHttpQuery.js:187:30)"
          ]
        }
      }
    }
  ]
}
```
any idea on what to do here?
is it possible to support those operators for String fields?

@neo4j/graphql version: 3.2.3
Node version: v17.3.0
Neo4j version: 4.4.3

**Comment by darrellwarde** [comment]
I have to ask, why are you not using `DateTime` for this field? It supports the comparison operators you are after.

**Comment by noureldin-khaled** [comment]
@darrellwarde I tried using `DateTime` like so
```graphql
type Address{
    uuid: String
    full_address: String
    city: String
    state: String
    zip: String
    creation_date: DateTime
}
```
But it now gives me an error when trying to retrieve the creation date
```graphql
query Addresses {
  addresses {
    uuid
    full_address
    creation_date
  }
}
```
gives
```
{
  "errors": [
    {
      "message": "Failed to invoke function `apoc.date.convertFormat`: Caused by: java.time.format.DateTimeParseException: Text '2022-05-23 17:57:25' could not be parsed at index 10",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "addresses"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "code": "Neo.ClientError.Procedure.ProcedureCallFailed",
          "name": "Neo4jError",
          "stacktrace": [
            "Neo4jError: Failed to invoke function `apoc.date.convertFormat`: Caused by: java.time.format.DateTimeParseException: Text '2022-05-23 17:57:25' could not be parsed at index 10",
            "",
            "    at captureStacktrace (/var/www/graphql.zlien.com/node_modules/neo4j-driver-core/lib/result.js:239:17)",
            "    at new Result (/var/www/graphql.zlien.com/node_modules/neo4j-driver-core/lib/result.js:59:23)",
            "    at newCompletedResult (/var/www/graphql.zlien.com/node_modules/neo4j-driver-core/lib/transaction.js:375:12)",
            "    at Object.run (/var/www/graphql.zlien.com/node_modules/neo4j-driver-core/lib/transaction.js:229:20)",
            "    at Transaction.run (/var/www/graphql.zlien.com/node_modules/neo4j-driver-core/lib/transaction.js:101:34)",
            "    at execute (/var/www/graphql.zlien.com/node_modules/@neo4j/graphql/dist/utils/execute.js:87:51)",
            "    at resolve (/var/www/graphql.zlien.com/node_modules/@neo4j/graphql/dist/schema/resolvers/query/read.js:33:57)",
            "    at Object.addresses (/var/www/graphql.zlien.com/node_modules/@neo4j/graphql/dist/schema/resolvers/wrapper.js:73:12)",
            "    at field.resolve (/var/www/graphql.zlien.com/node_modules/apollo-server-core/dist/utils/schemaInstrumentation.js:56:26)",
            "    at executeField (/var/www/graphql.zlien.com/node_modules/graphql/execution/execute.js:481:20)"
          ]
        }
      }
    }
  ],
  "data": null
}
```

I'm guessing it because the creation date is stored in the format `yyyy-mm-dd H:m:s` which is not the format passed to `apoc.date.convertFormat`, is there a way to tell it what is the current format so that it parses it correctly?

Also, I tried using it in a where condition (keep in mind I have 1 Address Node locally with `creation_date` of `2022-05-23 17:57:25`)
```graphql
query Addresses {
  addresses(where: { creation_date_GT: "2022-01-01 17:00:00" }) {
    uuid
    full_address
  }
}
```
gives
```graphql
{
  "data": {
    "addresses": []
  }
}
```
which is incorrect it should return the 1 node I have, so I'm assuming the condition is not being applied correctly.

**Comment by MacondoExpress** [comment]
I think this is closed by the #1694 ! 😃
